sql/src/main/java/io/crate/execution/engine/aggregation/GroupBySingleNumberCollector.java

```diff
@@ -131,21 +131,31 @@ private static void validateInputType(DataType valueType) {
          * The entry overhead is computed by observing the effect/key and /value on the used space when the capacity is
          * doubled eg. for Integer going from 2 items to 4 items yields an 8 byte increase in used space for both keys
          * and values (so an extra 4 byte / key and 4 byte / value) which results in a total of 8 bytes overhead/ entry.
+         *
+         * During the construction of the map we reserve space for 4 elements
+         * because of the defaults of 8 for initial capacity and 0.5 for load factor.
```

Comment:
if 8 is the initial capacity, doesn't that mean we need to account for the 8 empty slots?

Reply:
They it works is that it starts with maxSize = capacity * loadFactor = 4. When the 5th element is added it goes to 8 and so on....